feat: Configure ESLint and Prettier in both the repositories done (#6)

Diksha16110970 opened the pull request:
feat: Configure ESLint and Prettier in both the repositories done (#6)

Diksha16110970 committed:
feat: Configure ESLint and Prettier in both the repositories done (#6)

Diksha16110970 commented:
@rakeshtembhurne Sir n @AmanSurushe , I have raised the PR

AmanSurushe commented:
@Diksha16110970   #6 

🔍 Critical Issues to Address

  1. Broken ESLint Configuration - Frontend & Backend

  // Both frontend/eslint.config.mjs:18 and backend/eslint.config.mjs:20
  '@typescript-eslint': require('@typescript-eslint/eslint-plugin'), // ❌ CJS in ESM
  Impact: ESLint will fail to start
  Fix: Use proper ESM imports:
  import tsPlugin from '@typescript-eslint/eslint-plugin';
  // Then use: '@typescript-eslint': tsPlugin

  2. Missing Airbnb Dependency - Backend

  import airbnb from 'eslint-config-airbnb-extended'; // ❌ Not installed
  Impact: Build/lint failures
  Fix: Either install the package or remove the import

  3. Parser Configuration Issues

  - Frontend: Redundant parser declaration (already handled by Next.js config)
  - Backend: Manual parser config conflicts with typescript-eslint recommended setup

  ---
  ⚠️ Configuration Problems

  4. Inconsistent Rule Enforcement

  Rules changed from 'off' to 'error' without testing:
  - @typescript-eslint/no-unsafe-* rules now 'error' (previously 'off')
  - This could break existing code

  5. JSON Formatting Issues

  Prettier config files have inconsistent formatting:
  // Missing trailing comma, extra newlines
  {
  "singleQuote": true,  // ❌ Should be consistent
  "semi": true
  }

  6. Package.json Script Issues - Frontend

  "lint": "eslint \"{pages,components,src,app}/**/*.{ts,tsx,js,jsx}\" --fix"
  Problem: Hardcoded paths may not match actual project structure

  ---
  🔧 Code Quality Recommendations

  7. Overly Strict TypeScript Rules

  Current setup enables strict rules that might be too aggressive for this project phase:
  '@typescript-eslint/no-unsafe-assignment': 'error', // Consider 'warn'
  '@typescript-eslint/no-unsafe-call': 'error',       // Consider 'warn'

  8. Missing Configuration Validation

  No mechanism to validate ESLint/Prettier configs work before merge.

  ---
  ✅ Positive Aspects

  1. Comprehensive Scope: Addresses both frontend and backend
  2. Good File Structure: Proper .prettierrc and .prettierignore files
  3. Documentation Updates: README files updated appropriately
  4. Unused Imports Plugin: Good addition for code cleanliness
  5. Consistent Formatting Rules: Standardized across projects

  ---
  📋 Action Items for Diksha

  High Priority (Must Fix):

  1. Fix ESM/CJS mixing in ESLint configs
  2. Remove or properly install eslint-config-airbnb-extended
  3. Test that npm run lint and npm run format work in both projects
  4. Validate TypeScript compilation doesn't break

  Medium Priority (Should Fix):

  5. Standardize JSON formatting in .prettierrc files
  6. Consider relaxing some TypeScript rules from 'error' to 'warn'
  7. Verify frontend lint script paths match actual project structure

  Testing Required:

  8. Run npm install in both frontend and backend
  9. Execute npm run lint and npm run format in both projects
  10. Verify no breaking changes to existing code

CC @rakeshtembhurne

Diksha16110970 committed:
fix: Configure ESLint and Prettier in both the repositories (#6)

Diksha16110970 commented:
@AmanSurushe I have made the relevant changes; kindly review them.

AmanSurushe commented:
@Diksha16110970  #6 

 Before merging, verify these commands work:

  Backend

  cd backend
  npm install
  npm run lint
  npm run format
  npm run build 

  Frontend

  cd frontend
  npm install
  npm run lint
  npm run format
  npm run build

CC @rakeshtembhurne

Diksha16110970 committed:
refactor: Configure ESLint and Prettier in both the repositories (#6)

AmanSurushe committed:
Merge remote-tracking branch 'upstream/develop' into DEV_Diksha

AmanSurushe committed:
feat: standardize code quality tooling and documentation

- Upgrade backend Prettier to ^3.6.2 for version consistency
- Add lint:check and format:check scripts to backend package.json
- Create comprehensive .editorconfig for IDE consistency
- Update READMEs with detailed ESLint configuration documentation
- Document rationale for different ESLint configs (Next.js vs Airbnb)
- Fix all formatting issues across both repositories
- Ensure all lint and format commands pass successfully

AmanSurushe commented:
@Diksha16110970    

this can be merge now #6 

Cc @rakeshtembhurne

AmanSurushe reviewed (approved):
Approve
